AerospikeClient/Async/AsyncCommand.cs

```diff
@@ -229,29 +241,60 @@ private void ExecuteCommand()
 			}
 			catch (AerospikeException.Connection aec)
 			{
+				exception = aec;
 				ErrorCount++;
 				ConnectionFailed(aec);
 			}
 			catch (AerospikeException.Backoff aeb)
 			{
+				exception = aeb;
 				ErrorCount++;
 				Backoff(aeb);
 			}
 			catch (AerospikeException ae)
 			{
+				exception = ae;
 				ErrorCount++;
 				FailOnApplicationError(ae);
 			}
 			catch (SocketException se)
 			{
+				exception = se;
 				ErrorCount++;
 				OnSocketError(se.SocketErrorCode);
 			}
 			catch (Exception e)
 			{
+				exception = e;
 				ErrorCount++;
 				FailOnApplicationError(new AerospikeException(e));
 			}
+			finally
+			{
+				if (activity != null && exception != null)
+				{
+					activity.SetStatus(ActivityStatusCode.Error, exception.ToString());
```

Comment:
I don't think the status description should contain the full exception. There are exception tags made for this https://opentelemetry.io/docs/specs/semconv/attributes-registry/exception. See how they do it in `OpenTelemetry.Api`: https://github.com/open-telemetry/opentelemetry-dotnet/blob/14ee5c456ee91ffc11e29b35333834b2e04f53d3/src/OpenTelemetry.Api/Trace/TelemetrySpan.cs#L266.

That would allow a UI to format the exception if the rights tags are present.